glue/plugins/scatter_group/client.py

```diff
@@ -0,0 +1,239 @@
+            to_comp = coerce_numeric(np.array(range(0, l)))
+            to_comp.setflags(write=False)
+            none_comp._data = to_comp
+
+        comp = data.components if show_hidden else data.visible_components
+        groups = [comp[-1]]
+        for c in comp:
+            if data.get_component(c).group:
+                groups.append(c)
+        return groups
+
+    def add_layer(self, layer):
+        """ Adds a new visual layer to a client, to display either a dataset
+        or a subset. Updates both the client data structure and the
+        plot.
+
+        Returns the created layer artist
+
+        :param layer: the layer to add
+        :type layer: :class:`~glue.core.data.Data` or :class:`~glue.core.subset.Subset`
+        """
+        if layer.data not in self.data:
+            raise TypeError("Layer not in data collection")
+        if layer in self.artists:
+            return self.artists[layer][0]
+
+        result = ScatterGroupLayerArtist(layer, self.axes)
+        self.artists.append(result)
+        self._update_layer(layer)
+        self._ensure_subsets_added(layer)
+        return result
+
+    def _set_xydata(self, coord, attribute, snap=True):
+        """ Redefine which components get assigned to the x/y axes
+
+        :param coord: 'x' or 'y'
+           Which axis to reassign
+        :param attribute:
+           Which attribute of the data to use.
+        :type attribute: core.data.ComponentID
+        :param snap:
+           If True, will rescale x/y axes to fit the data
+        :type snap: bool
+        """
+
+        if coord not in ('x', 'y', 'g'):
+            raise TypeError("coord must be one of x, y, g")
+        if not isinstance(attribute, ComponentID):
+            raise TypeError("attribute must be a ComponentID")
+
+        # update coordinates of data and subsets
+        if coord == 'x':
+            new_add = not self._xset
+            self.xatt = attribute
+            self._xset = self.xatt is not None
+        elif coord == 'y':
+            new_add = not self._yset
+            self.yatt = attribute
+            self._yset = self.yatt is not None
+        elif coord == 'g':
+             self.gatt = attribute
+             self._gset = self.gatt is not None
+
+        # update plots
+        list(map(self._update_layer, self.artists.layers))
+
+        if coord == 'x' and snap:
+            self._snap_xlim()
+            if new_add:
+                self._snap_ylim()
+        elif coord == 'y' and snap:
+            self._snap_ylim()
+            if new_add:
+                self._snap_xlim()
+
+        self._update_axis_labels()
+        self._pull_properties()
+        self._redraw()
+
+    def restore_layers(self, layers, context):
+        """ Re-generate a list of plot layers from a glue-serialized list"""
+        for l in layers:
+            cls = lookup_class(l.pop('_type'))
+            if cls != ScatterGroupLayerArtist:
```

Comment:
Same comment here, if we lookup which class to compare to using an attribute, then we don't need to modify this function in your new viewer